action.yml

```diff
@@ -35,28 +34,66 @@ runs:
       with:
         cmd: yq '${{ inputs.image-name-key-path }}' '${{ inputs.metadata-file }}'
 
+    - name: Parse organization
+      id: parse-org
+      shell: bash
+      env:
+        CALLING_ORGANIZATION: ${{ github.event.organization.login }}
+        INPUT_ORGANIZATION: ${{ inputs.organization }}
+      run: echo "org=${INPUT_ORGANIZATION:-$CALLING_ORGANIZATION}" >> "$GITHUB_OUTPUT"
+
+    # Take this path if the event is a branch deletion
+    - if: github.event_name == 'delete'
+      name: Delete matching docker tags
+      uses: actions/github-script@v7
+      env:
+        ORGANIZATION: ${{ steps.parse-org.outputs.org }}
+        IMAGE_NAME: ${{ steps.yaml-data.outputs.result }}
+      with:
+        script: |
+          const script = require(`${process.env['GITHUB_ACTION_PATH']}/delete-tags.js`)
+          await script({ github, context, core })
+
+    # Take this path if the event is not a deletion
     - name: Create tags
+      if: github.event_name != 'delete'
       id: meta
       uses: docker/metadata-action@v5
       with:
         flavor: |
           latest=false
-        images: ${{ inputs.registry }}/${{ steps.yaml-data.outputs.result }}
+        images: ghcr.io/${{ steps.parse-org.outputs.org }}/${{ steps.yaml-data.outputs.result }}
```

Comment:
**question (non-blocking)**: Is there a reason to switch to a hard-coded registry?

Reply:
Yash, it's like you don't even _read_ all of my errata! Unbelievable.

Yes, the tag-deletion logic specifically requires using the [GitHub API](https://docs.github.com/en/rest/packages/packages?apiVersion=2022-11-28#delete-package-version-for-an-organization), so it's not applicable to other package registries.

Theoretically we could add support for DockerHub/ECR/whatever, but those would be completely different implementations and I didn't think it was a necessary feature.

Reply:
S'okay, I just realized I've been using the word completely wrong anyway. I've been using it like "footnotes" or "annotations", but those don't sound _nearly_ fancy enough. I'll figure out something suitably sesquipedalian for my next PR.